python/ray/util/ml_utils/filelock.py

```diff
@@ -1,8 +1,14 @@
-from filelock import FileLock
-from pathlib import Path
 import hashlib
 import os
+import sys
 import tempfile
+from pathlib import Path
+
+if sys.platform == "win32":
+    from filelock import WindowsFileLock as FileLock
+else:
+    # Alias to UnixFileLock.
+    from filelock import FileLock as FileLock
```

Comment:
Remove filelock.

Reply:
Actually, this change is irrelevant to this PR. It is a bug we don't use WindowsFileLock on Windows I believe. If you want me to separate this out to a separate PR, I can do that.

Reply:
Yeah, this is a cherry-pick PR. Let's keep it minimal.